doc/Type/Mu.pod6

```diff
@@ -343,86 +343,9 @@ case we are declaring this C<new> method simply to avoid the extra syntax of
 using pairs when creating the object. C<self.bless> returns the object, which is
 in turn returned by C<new>.
 
-However, in general, implementing a customized C<new> method might not be the best
-way of initializing a class, even more so if the default constructor is
-disabled, since it can make it harder to correctly initialize the class from a
-subclass. For instance, in the above example, the C<new> implementation takes
-two positional arguments that must be passed from the subclass to the superclass
-in the exact order. That is not a real problem if it's documented, but take
-into account C<bless> will eventually be calling C<BUILD>
-in the class that is being instantiated. This might result in some unwanted
-problems, like having to create a C<BUILD> submethod to serve it correctly:
-
-=begin code
-class Point {
-    has Int $.x;
-    has Int $.y;
-    multi method new($x, $y) {
-        self.bless(:$x, :$y);
-    }
-}
-
-class Point-with-ID is Point {
-    has Int $.ID  is rw = 0;
-
-    submethod BUILD( *%args ) {
-        say %args;                # OUTPUT: «{x => 1, y => 2}␤»
-        for self.^attributes -> $attr {
-            if $attr.Str ~~ /ID/ {
-                $attr.set_value( self, "*" ~ %args<x> ~ "-" ~ %args<y> ) ;
-            }
-        }
-    }
-}
-
-my $p = Point-with-ID.new(1,2);
-say $p.raku;
-# OUTPUT: «Point-with-ID.new(ID => "*1-2", x => 1, y => 2)␤»
-=end code
-
-In this code, C<bless>, called within C<Point.new>, is eventually calling
-C<BUILD> with the same parameters. We have to create a convoluted way of using
-the C<$.ID> attribute using the metaobject protocol so that we can instantiate
-it and thus serve that C<new> constructor, which can be called on
-C<Point-with-ID> since it is a subclass.
-
-We might have to use something similar if we want to instantiate superclasses.
-C<bless> will help us with that, since it is calling across all the hierarchy:
-
-=begin code
-class Str-with-ID is Str {
-    my $.counter = 0;
-    has Int $.ID  is rw = 0;
-
-    multi method new( $str ) {
-        self.bless( value => $str, ID => $.counter++ );
-    }
-
-    submethod BUILD( *%args ) {
-        for self.^attributes -> $attr {
-            if $attr.Str ~~ /ID/ {
-                $attr.set_value( self, %args<ID> ) ;
-            }
-        }
-    }
-}
-
-say Str-with-ID.new("1.1,2e2").ID;                  # OUTPUT: «0␤»
-my $enriched-str = Str-with-ID.new("3,4");
-say "$enriched-str, {$enriched-str.^name}, {$enriched-str.ID}";
-# OUTPUT: «3,4, Str-with-ID, 1␤»
-=end code
-
```

Comment:
And then, of course, there's this whole thing. Using the same (probably wrong) technique in `BUILD` but, again, consequence of #2077 and serves its purpose. So if you want, you can simply change this `BUILD` to the new version that simply assigns a value to ID.

Reply:
I have re-added the example, simplified it a bit and adapted the explanation accordingly.